src/chain_spec.rs

```diff
@@ -411,396 +381,636 @@ macro_rules! protocol_fee {
-
-    AlcyoneConfig::GenesisConfig {
-        frame_system: Some(AlcyoneConfig::SystemConfig {
-            code: alcyone::WASM_BINARY
-                .expect("WASM binary was not generated")
-                .to_vec(),
-            changes_trie_config: Default::default(),
-        }),
-        asset: Some(asset!()),
-        pallet_checkpoint: Some(checkpoint!()),
-        identity: Some(AlcyoneConfig::IdentityConfig {
-            identities: all_identities,
-            secondary_keys,
-            ..Default::default()
-        }),
-        balances: Some(AlcyoneConfig::BalancesConfig {
-            balances: balances(&initial_authorities, &endowed_accounts),
-        }),
-        bridge: Some(AlcyoneConfig::BridgeConfig {
-            admin: seeded_acc_id("polymath_1"),
-            creator: seeded_acc_id("polymath_1"),
-            signatures_required: 3,
-            signers: bridge_signers(),
-            timelock: alcyoneTime::MINUTES * 15,
-            bridge_limit: (30_000_000_000, alcyoneTime::DAYS),
-        }),
-        pallet_indices: Some(AlcyoneConfig::IndicesConfig { indices: vec![] }),
-        pallet_sudo: Some(AlcyoneConfig::SudoConfig { key: root_key }),
-        pallet_session: Some(session!(
-            AlcyoneConfig,
-            initial_authorities,
-            alcyone_session_keys
-        )),
-        pallet_staking: Some(staking!(initial_authorities, stakers, PerThing::zero())),
-        pallet_pips: Some(AlcyoneConfig::PipsConfig {
-            prune_historical_pips: false,
-            min_proposal_deposit: 0,
-            default_enactment_period: alcyoneTime::DAYS * 7,
-            max_pip_skip_count: 1,
-            active_pip_limit: 1000,
-            pending_pip_expiry: <_>::default(),
-        }),
-        pallet_im_online: Some(im_online!()),
-        pallet_authority_discovery: Some(Default::default()),
-        pallet_babe: Some(Default::default()),
-        pallet_grandpa: Some(Default::default()),
-        pallet_contracts: Some(AlcyoneConfig::ContractsConfig {
-            current_schedule: contracts::Schedule {
-                enable_println, // this should only be enabled on development chains
+pub mod alcyone_testnet {
+    use super::*;
+    use polymesh_runtime_testnet::{self as rt, config, constants::time};
+
+    pub type ChainSpec = sc_service::GenericChainSpec<config::GenesisConfig>;
+
+    fn session_keys(
+        grandpa: GrandpaId,
+        babe: BabeId,
+        im_online: ImOnlineId,
+        authority_discovery: AuthorityDiscoveryId,
+    ) -> rt::SessionKeys {
+        rt::SessionKeys {
+            babe,
+            grandpa,
+            im_online,
+            authority_discovery,
+        }
+    }
+
+    fn genesis(
+        initial_authorities: Vec<InitialAuth>,
+        root_key: AccountId,
+        endowed_accounts: Vec<AccountId>,
+        enable_println: bool,
+    ) -> config::GenesisConfig {
+        let init_ids = [
```

Comment:
It looks like some parts of genesis fn can be also dedup.

Reply:
I expect this will diverge over time so deduping this bit doesn't seem good atm.